llama-index-integrations/readers/llama-index-readers-azstorage-blob/llama_index/readers/azstorage_blob/base.py

```diff
@@ -42,34 +42,18 @@ class AzStorageBlobReader(BaseReader):
             The credentials with which to authenticate. This is optional if the account URL already has a SAS token.
     """
 
-    def __init__(
-        self,
-        *args: Any,
-        container_name: str,
-        prefix: str = "",
-        blob: Optional[str] = None,
-        name_starts_with: Optional[str] = None,
-        include: Optional[Any] = None,
-        file_extractor: Optional[Dict[str, Union[str, BaseReader]]] = None,
-        connection_string: Optional[str] = None,
-        account_url: Optional[str] = None,
-        credential: Optional[Any] = None,
-        **kwargs: Any,
-    ) -> None:
-        """Initializes Azure Storage Account."""
-        super().__init__(*args, **kwargs)
-
-        self.container_name = container_name
-        self.prefix = prefix
-        self.connection_string = connection_string
-        self.blob = blob
-        self.name_starts_with = name_starts_with
-        self.include = include
-        self.file_extractor = file_extractor
-        self.account_url = account_url
-        self.credential = credential
-        # Not in use. As part of the TODO below. Is part of the kwargs.
-        # self.preloaded_data_path = kwargs.get('preloaded_data_path', None)
+    container_name: str
```

Comment:
why moving those out of init function? will it introduce some break changes

Reply:
oh i think this is moved out because its now a subclass of Pydantic `BaseModel`. So we can define them as Fields, and leverage Pydantic's built in construction process.

Reply:
Shouldn't have any breaking changes, I think.